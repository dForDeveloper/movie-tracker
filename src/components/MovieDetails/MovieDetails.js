import React from 'react';
import PropTypes from 'prop-types';

const MovieDetails = (props) => {
  const { title, release_date, poster_path, vote_average, overview } = props;
  return (
    <div className="MovieDetails">
      <img
        src={`https://image.tmdb.org/t/p/w500${poster_path}`}
        alt={title}
        className="movie-image"
      />
      <div className="movie-info">
        <h3 className="movie-title">{title}</h3>
          <p className="movie-overview"> {overview}</p>
          <p className="movie-details">
            <span className="detail-bold">
              Release Date: </span>
            {release_date}
          </p>
          <p className="movie-details">
            <span className="detail-bold">
              Rating: </span>
            {vote_average}
          </p>
      </div>
    </div>
  )
}

export default MovieDetails;

MovieDetails.propTypes = {
  adult: PropTypes.bool,
  backdrop_path: PropTypes.string,
  genre_ids: PropTypes.array,
  id: PropTypes.number,
  original_language: PropTypes.string,
  original_title: PropTypes.string,
  overview: PropTypes.string,
  popularity: PropTypes.number,
  poster_path: PropTypes.string,
  release_date: PropTypes.string,
  title: PropTypes.string,
  video: PropTypes.bool,
  vote_average: PropTypes.number,
  vote_count: PropTypes.number
}