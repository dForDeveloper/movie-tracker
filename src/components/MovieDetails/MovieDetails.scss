.MovieDetails {
  display: grid;
  justify-content: center;
}

.movie-title {
  text-align: center;
}

.movie-image {
  height: 600px;
}