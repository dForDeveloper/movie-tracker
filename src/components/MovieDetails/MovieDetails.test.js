import React from 'react';
import { shallow } from 'enzyme';
import MovieDetails from './MovieDetails.js';

describe('MovieDetails', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <MovieDetails />
    );
  });

  it('should properly render the component elements', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
