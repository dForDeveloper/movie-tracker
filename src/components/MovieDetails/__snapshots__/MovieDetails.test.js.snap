// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieDetails component should properly render the component elements 1`] = `
<div
  className="MovieDetails"
>
  <h3
    className="h3"
  />
  <img
    className="MovieDetails--image"
    src="https://image.tmdb.org/t/p/w500undefined"
  />
  <p />
  <p>
    Release Date: 
  </p>
  <p>
    Rating: 
  </p>
</div>
`;
